src/main/java/qupath/bioimageio/spec/tensor/Processing.java

```diff
@@ -0,0 +1,397 @@
+         * | per_sample  | Compute for each sample individually |
+         */
+        public ProcessingMode getMode() {
+            return mode;
+        }
+
+        /**
+         * The subset of axes to scale jointly.
+         * For example xy to normalize the two image axes for 2d data jointly.
+         * Default: scale all non-batch axes jointly.
+         * @return the subset of axes to scale jointly.
+         */
+        public Axis[] getAxes() {
+            return axes.clone();
+        }
+
+    }
+
+    /**
+     * A scaling operation to the mean and variance of a reference tensor.
+     */
+    public static class ScaleMeanVariance extends ProcessingWithMode {
+
+        String referenceTensor;
+
+        ScaleMeanVariance() {
+            super("scale_mean_variance");
+        }
+
+        /**
+         * The tensor to match mean and variance to.
+         */
+        public String getReferenceTensor() {
+            return referenceTensor;
+        }
+    }
+
+    /**
+     * A scaling operation to a reference tensor.
+     */
+    public static class ScaleRange extends ProcessingWithMode {
+
+        String referenceTensor; // TODO: Figure out whether to use this somehow
+
+        double minPercentile = 0.0;
+        double maxPercentile = 100.0;
+
+        ScaleRange() {
+            super("scale_range");
+        }
+
+        /**
+         * The minimum percentile.
+         * @return the min percentile
+         */
+        public double getMinPercentile() {
+            return minPercentile;
+        }
+        /**
+         * The maximum percentile.
+         * @return the max percentile
+         */
+        public double getMaxPercentile() {
+            return maxPercentile;
+        }
+
+        /**
+         * The reference tensor, which we use for range scaling.
+         * @return the reference tensor
+         */
+        public String getReferenceTensor() {
+            return referenceTensor;
+        }
+
+
+    }
+
+    /**
+     * A scaling operation to zero mean and unit variqnce.
+     */
+    public static class ZeroMeanUnitVariance extends ProcessingWithMode {
+
+        double[] mean;
+        double[] std;
+
+        ZeroMeanUnitVariance() {
+            super("zero_mean_unit_variance");
+        }
+
+        /**
+         * The mean value(s) to use for `mode: fixed`.
+         * For example `[1.1, 2.2, 3.3]` in the case of a 3 channel image with `axes: xy`.
+         * @return the mean value(s)
+         */
+        public double[] getMean() {
+            return mean == null ? null : mean.clone();
+        }
+
+        /**
+         * The standard deviation values to use for `mode: fixed`. Analogous to mean.
+         * @return the std values.
+         */
+        public double[] getStd() {
+            return std == null ? null : std.clone();
+        }
+
+    }
+
+    // todo: this shouldn't be public, but it's in a different package
```

Comment:
@petebankhead this is one example - the deserialization code for this class is now in a different package to the model class, where the main parsing logic exists